src/libawkward/array/NumpyArray.cpp

```diff
@@ -651,16 +651,58 @@ namespace awkward {
     }
   }
 
+  const std::vector<ssize_t> flatten_shape(const std::vector<ssize_t>& shape, int64_t axis) {
+    if (shape.size() == 1) {
+      return std::vector<ssize_t>();
+    }
+    else {
+      std::vector<ssize_t> out;
+      const auto& indx = begin(shape) + axis;
```

Comment:
32-bit vs 64-bit error here

Reply:
fixed in https://github.com/scikit-hep/awkward-1.0/pull/98